### DISCUSSION TITLE
⚡ Xcode 16.1

### DISCUSSION THREAD

**Discussion by helinanever** [codemagic-ci-cd/codemagic-docs#2685] [Announcements]
**Xcode 16.1 Beta 1** is now available in Codemagic. This image comes with runtimes for iOS 18.1, watchOS 11.0, tvOS 18.0, and visionOS 2.0. To use it, set the Xcode version to `16.1` or `edge` in your build configuration.

See details about the build machine specs and preinstalled software in our [documentation](https://docs.codemagic.io/specs/versions-macos/).

**Comment by helinanever** [comment]
**Xcode 16.1 Beta 2** is now available on the build machines.

**Comment by helinanever** [comment]
This update is a little late but **Xcode 16.1 Beta 3** is available on the build machines as of October 9.

**Comment by helinanever** [comment]
**Xcode 16.1 RC 1** is now available on Codemagic build machines.

**Comment by helinanever** [comment]
The 16.1 RC 1 build has been promoted to **Xcode 16.1 Release**. You can use this version by selecting `16.1` in your Xcode version settings.

**Comment by helinanever** [comment]
As we approach the holiday season, ensuring the stability of our build environment is our top priority. After careful consideration, we have decided to delay (and potentially skip) making Xcode 16.1 machines default. We will maintain Xcode 16.0 as the default version for now to minimize potential disruptions and are evaluating a direct transition to Xcode 16.2 in the future.

To use Xcode 16.1 machines, specify `16.1` as the Xcode version in your workflow configuration.

**Reply by habyaad** [reply]
little wonder!
